src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java

```diff
@@ -99,17 +103,43 @@ public boolean validCheckForObject(final AtlasObject object)
         // Create a ComplexRoundabout based on object
         final ComplexRoundabout complexRoundabout = new ComplexRoundabout((Edge) object,
                 this.leftDrivingCountries);
-        // If the ComplexRoundbaout is invalid add the reasons why to the instructions, so that it
-        // will be flagged
-        if (!complexRoundabout.isValid())
-        {
-            instructions.addAll(complexRoundabout.getAllInvalidations().stream()
-                    .map(ComplexEntity.ComplexEntityError::getReason).collect(Collectors.toSet()));
-        }
         // Get the roundabout Edges
         final Set<Edge> roundaboutEdgeSet = complexRoundabout.getRoundaboutEdgeSet();
         // Get the roundabout Route
         final Route roundaboutEdges = complexRoundabout.getRoundaboutRoute();
+
+        // If the ComplexRoundabout is invalid add the reasons why to the instructions, so that it
+        // will be flagged
+        if (!complexRoundabout.isValid())
+        {
+            // AutoFix candidate only for wrong direction case.
+            if (complexRoundabout.getAllInvalidations().size() == 1 && complexRoundabout
+                    .getAllInvalidations().get(0).getReason().equals(WRONG_WAY_INSTRUCTIONS))
+            {
+                // Mark that the Edges have been processed
+                roundaboutEdgeSet.forEach(
+                        roundaboutEdge -> this.markAsFlagged(roundaboutEdge.getIdentifier()));
+
+                return Optional.of(this
+                        .createFlag(roundaboutEdgeSet,
+                                this.getLocalizedInstruction(1, object.getOsmIdentifier()))
+                        .addFixSuggestion(FeatureChange.add(
```

Comment:
Sorry, I misread this slightly the first time. As well as flagging each edge in roundaboutEdgeSet, as you have now done, each of those edges also needs its own fix suggestion to reverse the geometry.